src/inspect.mjs

```diff
@@ -1,95 +1,489 @@
+  if (type !== constructor) {
+    if (constructor !== null) {
+      base += ` (${constructor})`;
+    } else {
+      base += ' (null prototype)';
     }
-    return s;
   }
-  return '';
+
+  const os = ctx.stylize;
+  ctx.stylize = stylizeNoColor;
+  base += `: ${innerInspect(value)}]`;
+  ctx.stylize = os;
+
+  if (tag !== '' && tag !== constructor) {
+    base += ` [${tag}]`;
+  }
+
+  return ctx.stylize(base, type.toLowerCase());
 };
 
-const compactObject = (realm, value) => {
+const formatPropertyDescriptor = (desc, ctx, i) => {
+  Assert(Type(desc) === 'Descriptor');
+  if (IsDataDescriptor(desc)) {
+    return i(desc.Value);
+  } else {
+    Assert(IsAccessorDescriptor(desc));
+    let label;
+    if (desc.Get !== Value.undefined) {
+      label = desc.Set === Value.undefined ? 'Getter' : 'Getter/Setter';
+    } else {
+      label = 'Setter';
+    }
+    return (ctx.stylize(`[${label}]`, 'special'));
+  }
+};
+
+/* eslint-disable no-control-regex */
+const strEscapeSequencesReplacer = /[\x00-\x1f\x27\x5c\x7f-\x9f]/g;
+const strEscapeSequencesReplacerIgnoreQuotes = /[\x00-\x1f\x5c\x7f-\x9f]/g;
+/* eslint-enable no-control-regex */
+
+// Escaped control characters (plus the single quote and the backslash):
+const strEscapeArray = [
+  /* x00 - x07: */ '\\x00', '\\x01', '\\x02', '\\x03', '\\x04', '\\x05', '\\x06', '\\x07',
+  /* x08 - x0F: */ '\\b', '\\t', '\\n', '\\x0B', '\\f', '\\r', '\\x0E', '\\x0F',
+  /* x10 - x17: */ '\\x10', '\\x11', '\\x12', '\\x13', '\\x14', '\\x15', '\\x16', '\\x17',
+  /* x18 - x1F: */ '\\x18', '\\x19', '\\x1A', '\\x1B', '\\x1C', '\\x1D', '\\x1E', '\\x1F',
+  /* x20 - x2F: */ '', '', '', '', '', '', '', "\\'", '', '', '', '', '', '', '', '',
+  /* x30 - x3F: */ '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '',
+  /* x40 - x4F: */ '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '',
+  /* x50 - x5F: */ '', '', '', '', '', '', '', '', '', '', '', '', '\\\\', '', '', '',
+  /* x60 - x6F: */ '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '',
+  /* x70 - x7F: */ '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '\\x7F',
+  /* x80 - x87: */ '\\x80', '\\x81', '\\x82', '\\x83', '\\x84', '\\x85', '\\x86', '\\x87',
+  /* x88 - x8F: */ '\\x88', '\\x89', '\\x8A', '\\x8B', '\\x8C', '\\x8D', '\\x8E', '\\x8F',
+  /* x90 - x97: */ '\\x90', '\\x91', '\\x92', '\\x93', '\\x94', '\\x95', '\\x96', '\\x97',
+  /* x98 - x9F: */ '\\x98', '\\x99', '\\x9A', '\\x9B', '\\x9C', '\\x9D', '\\x9E', '\\x9F',
+];
+
+const escapeFn = (str) => strEscapeArray[str.charCodeAt(0)];
+
+// Roughly based on the JSON stringify escaping.
+const strEscape = (str) => {
+  let replacer = strEscapeSequencesReplacer;
+  let quote = 0x27;
+
+  if (str.includes("'")) {
+    if (!str.includes('"')) {
+      // If the string contains single quotes and not double quotes,
+      // then we wrap it in double quotes:
+      quote = -1;
+    } else if (!str.includes('`') && !str.includes('${')) {
```

Comment:
i don't think we should ever format a string with backticks. I would say choose `'` or `"` based on the contents and fall back to `'` if neither one is best.